app/assets/javascripts/analytics.js.erb

```diff
@@ -2,112 +2,87 @@
-
-  // Fire an event whenever a user marks a lesson uncompleted on the individual lesson page
-  $('.lc-end-wrapper').on('click', '.lc-uncomplete-link', function(e) {
-    _gaq.push(['_trackEvent', 'lesson_completion', 'mark_not_completed', window.location.href, 1]);
-  });
-
-  // Fire an event whenever a user tries to mark a lesson uncompleted on the individual lesson page by clicking the checkbox icon (which isn't a link)
-  $('.lc-end-wrapper').on('click', '.checkbox-container.lc-checked', function(e) {
-    _gaq.push(['_trackEvent', 'lesson_completion', 'click_not_linked_checkbox', 'lesson_page', 1]);
-  });
-
-  // Fire an event when a user clicks on the previous lesson navigation link in the individual lesson page
-  $('.lnav').on('click', '.lnav-prev-icon', function(e) {
-    _gaq.push(['_trackEvent', 'lesson_navigation', 'click_previous_lesson_link', 'lesson_page', 1]);
-  });
-
-  // Fire an event when a user clicks on the next lesson navigation link in the individual lesson page
-  $('.lnav').on('click', '.lnav-next-icon', function(e) {
-    _gaq.push(['_trackEvent', 'lesson_navigation', 'click_next_lesson_link', 'lesson_page', 1]);
+//Turbolinks magic happens here...
+document.addEventListener("turbolinks:load", function () {
+
+  $(function() {
+
+    // Fire a custom analytics event whenever any link is clicked
+    $('a').on('click', function(e) {
+      var $target = $(e.target);
+      var href = $target.attr('href');
+
+      // manually set href for elements wrapped in anchor tags
+      // this is because the click event target is actually the
+      // internal element, e.g. an <h3> tag wrapped in <a> tags
+      // will register a click target of <h3> not <a> so the
+      // 'href' property will be undefined and we must check its
+      // parents to acquire the href.
+      if(!href) {
+        href = $target.parents('a').attr('href');
+      }
+
+      // Send separate events for internal and external link clicks
+      var regexp = /^https?:/;
+      if(regexp.test(href)) {
+        _gaq.push(['_trackEvent', 'external_link', 'click', href, 1]);
+      } else {
+        _gaq.push(['_trackEvent', 'internal_link', 'click', href, 1]);
+      }
+    });
+
+    // Fire an event whenever a user marks a lesson completed on the lessons index page
+    // else whenever a student marks a lesson not completed on the lessons index page
+    $('[id^=section-lessons__]').on('click', function(e) {
+      if ($(this).children(":first").data('method') === 'post') {
+        _gaq.push(['_trackEvent', 'lesson_completion', 'mark_completed', $(this).children(":first")[0].href, 1]);
+      } else {
+        _gaq.push(['_trackEvent', 'lesson_completion', 'mark_not_completed', $(this).children(":first")[0].href, 1]);
+      }
+    });
+
+    //see_lesson_buttons.html.erb for lesson button analytics
+
+    // Fire an event when a user clicks on the gitter chat link in the individual lesson page
+    $('#gitter-button').on('click', function(e) {
```

Comment:
If we use a attribute selector for the href here we could cover all the gitter buttons on the site as well as the chat link in the navbar with one tracking event. The only chat link that would need a separate tracking event is the floating gitter button.

Heres the attribute selector we could use to select them
`[href="https://gitter.im/TheOdinProject/theodinproject"]`